Customize the color of highlighted paths.

varvaratou opened the pull request:
When calling the API method parcoords.highlight(data) the polyline paths corresponding to the passed array of data get highlighted. The highlighted items get the color of the foreground items. 
There may be cases when we want to assign a different color to the highlighted items. This PR provides this functionality by adding an API method:
_parcoords.highlightColor(color)_
If color is a string, highlighted polylines will be rendered as that color. If color is a function, that function will be run for each data element and the polyline color will be the return value. 

Bug Fix:
Currently every time we reorder and flip axis the highlighted items are cleared out. This PR fixes this behavior by restoring the highlighted paths after these operations.

This PR is supporting the work for [Frac 457](https://jira.autodesk.com/browse/FRAC-538)

Reviewer:
@Dewb

varvaratou-adsk committed:
Set coloring for path highlight

varvaratou-adsk committed:
When highlighting a path, preserve previously highlighted paths to true if 'keepHighlights' argument set to true

varvaratou-adsk committed:
Preserve highlight when reordering and flipping axis

Commit:
Concatenate d3.parcoords.js via make

varvaratou-adsk committed:
Typo fix: moved color setting to the correct function.

-Reveresed path_foreground() to its form before commit
0ec0a4a404fa9817fe923446ffa5c9ae3361b5d1
-Updated path_highlight to take into account customized highlight color.

varvaratou-adsk committed:
Concatenate d3.parcoords.js via make

Dewb commented:
Is the added boolean parameter on `.highlight([data])` really necessary? Since `.highlight()` with no arguments returns the existing highlight data, can't the client just do `pc.highlight(pc.highlight().concat(newHighlights))`?

varvaratou-adsk committed:
Revert pc.highlight(data) to its original version before commit d54d575

varvaratou-adsk committed:
Update README to include new API parcoords.highlightColor(color)

varvaratou-adsk committed:
Update README to include new API parcoords.highlightColor(color)

varvaratou-adsk committed:
Concatenate d3.parcoords.js via make

varvaratou commented:
@Dewb Update regarding comment about .highlight(data):
I removed the keepHighlights boolean option and restored  .highlight(data) to its original version.
Data can be indeed concatenated from the client and passed to the function. Coloring the highlights with different color or colors can be handled by just passing a function instead of a fixed string color to the new API method .highlightColor(color). So .highlight(data) does not need to change at all; this solution seems more generic and consistent to the way the library handles similar cases.

I updated README and PR description accordingly.

Dewb commented:
LGTM!
